### DISCUSSION TITLE
Filtering requests

### DISCUSSION THREAD

**Discussion by anschl** [eclipse-tractusx/tractusx-edc#1228] [General]
I'm trying to use the filter feature present in some EDC calls, e.g.:
- [/v2/catalog/request](https://app.swaggerhub.com/apis/eclipse-tractusx-bot/tractusx-edc/0.6.0#/Catalog/requestCatalog)
- [/v2/transferprocesses/request](https://app.swaggerhub.com/apis/eclipse-tractusx-bot/tractusx-edc/0.6.0#/Transfer%20Process/queryTransferProcesses)

Even though I didn't find documentation about the filter feature I derived its structure from creating Contract Definitions:
```json
{
  ...
  "filterExpression": [
    {
      "@type": "Criterion",
      "operandLeft": "id",
      "operator": "=",
      "operandRight": "1"
    }
  ]
}
```

While I got it working for some calls, I have two questions regarding them:
1. Is it possible to filter a deep property? Something like `odrl:hasPolicy.id = "MQ...."`
2. Is there documentation about which filter operations are available?
3. How can I find out which properties I can filter by? I'm interested in filtering by `assetId` in [/v2/transferprocesses/request](https://app.swaggerhub.com/apis/eclipse-tractusx-bot/tractusx-edc/0.6.0#/Transfer%20Process/queryTransferProcesses), but it says `Error validating schema: Field assetId not found on type class org.eclipse.edc.connector.transfer.spi.types.TransferProcess`, even though the [source](https://github.com/eclipse-edc/Connector/blob/7f72e601b278a8700895199cefab447a1b9689a0/spi/control-plane/transfer-spi/src/main/java/org/eclipse/edc/connector/controlplane/transfer/spi/types/TransferProcess.java) mentions it.

**Comment by lgblaumeiser** [comment]
Closed as stale, please renew item if this is still an issue
